scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1038,3 +1081,33 @@ def execute(self, pipeline, done):
             logger.info("Execution interrupted")
             for p in processes:
                 p.terminate()
+
+    @staticmethod
+    def get_cmake_filter_stages(instance):
+        """ Analyze filter expressions from test yaml and decide if dts and/or kconfig based filtering will be needed."""
+        dts_required = False
+        kconfig_required = False
+        full_required = False
+        filter_stages = []
+
+        if instance.testsuite.filter:
+            expressions = instance.testsuite.filter.split()
+            # Remove logic related expressions used by expr_parser. Logic is evaluated in expr_parser.
+            expressions = [expression for expression in expressions if expression not in expr_parser.reserved.keys()]
+            for expression in expressions:
+                # all dt based expressions have the same prefix. Hence, expressions without it relate to kconfig.
+                if "dt_" in expression:
```

Comment:
Shouldn't these checks only be verifying that the expression starts with `dt_` or `CONFIG_`? Testing this in a python3 REPL:
```
>>> expression="hello_dt_world"
>>> "dt_" in expression
True
```
Given this, some variable like CMAKE_CACHE_CONFIG_VAR might fall into the `elif "CONFIG" in expression` block. I'm not sure if we have any filter variables that would cause this, just something to consider.